syntax = "proto3";

import "map_renderer.proto";
import "transport_router.proto";


package transport_catalogue_serialize;

message Stop {
    bytes nameStop = 1;
    double latitude = 2;
    double longitude = 3;
}

message Bus {
    bytes nameBus = 1;
    bool isRing = 2;
    repeated int32 ptrToStops = 3;
}

message DistanceBetweenStop {
    bytes firstStop = 1;
    bytes secondStop = 2;
    uint32 distance = 3;
}

message TC {
    repeated Stop stops = 1;
    repeated Bus buses = 2;
    repeated DistanceBetweenStop distBtwnStop = 3;
}

message TCFull {
    TC tc_catalog = 1;
    render_settings_serialize.RenderSet render = 2;
    router_serialize.Router router = 3;
}